<?xml version="1.0"?>
<!-- Note: this is an example launch file, do not modify. -->
<launch>
  <arg name="linear_offset" default="[0.0, 0.0, -0.06]"/>
  <arg name="reference_frame" default="base_link"/>
  <arg name="target_frame" default="arm_link_5"/>

  <group ns="poses_to_move">
    <!-- Start pose -->
    <node pkg="mcr_common_converters" type="transform_to_pose_converter_node"
          name="move_transform_to_pose_converter" output="screen">

      <!-- Reference frame from which the pose is transformed -->
      <param name="reference_frame" type="str" value="$(arg reference_frame)"/>
      <!-- Target frame to which the pose is transformed -->
      <param name="target_frame" type="str" value="$(arg target_frame)"/>

      <!-- Cycle rate of the node (in Hz) -->
      <param name="loop_rate" type="double" value="10"/>

      <remap from="~converted_pose" to="/linear_motions/start_pose"/>
    </node>

    <!-- Goal pose -->
    <node pkg="mcr_common_converters" type="pose_shifter_node"
          name="move_pose_shifter" output="screen">

      <!-- Linear offset in X, Y and Z (in meters) -->
      <rosparam param="linear_offset" subst_value="True">$(arg linear_offset)</rosparam>

       <remap from="~pose_in" to="/linear_motions/start_pose"/>
       <remap from="~pose_out" to="/linear_motions/goal_pose"/>
    </node>

    <node pkg="mcr_common_converters" type="pose_shifter_pipeline_node"
          name="move_coordinator" output="screen">

      <!-- the node cycle rate (in hz) -->
      <param name="loop_rate" type="double" value="30"/>

      <remap from="~event_in" to="/poses_to_move/event_in"/>
      <remap from="~event_out" to="/poses_to_move/event_out"/>

      <!-- To trigger components -->
      <remap from="~start_transformer"
             to="/poses_to_move/move_transform_to_pose_converter/event_in"/>
      <remap from="~start_pose_shifter"
             to="/poses_to_move/move_pose_shifter/event_in"/>

      <!-- To get feedback from components -->
      <remap from="~pose_shifter_status"
             to="/poses_to_move/move_pose_shifter/event_out"/>
    </node>
  </group>
</launch>